Upgrade to 1.0 guide for existing customers (#189)
okteto/docs: src/content/enterprise/install/upgrade.mdx

For example:
```console
$ helm repo update
$ helm upgrade okteto okteto/okteto -f config.yaml --namespace=okteto --version 0.9.x
```
*You can use `helm ls` to find the name of your release.*

Please review [the release notes](/docs/enterprise/install/releases/) before upgrading. New features, known issues, and configuration changes will be listed there.

## Upgrading to 1.x

> The Okteto chart 1.0 comes with two breaking changes due to the following changes:

Reviewer: Should it be `1.x` here as well?

```suggestion
The Okteto chart 1.x comes with two breaking changes due to the following changes:
```

> The Okteto chart 1.0 comes with two breaking changes due to the following changes:

Reviewer 2: ```suggestion
Okteto 1.0 has two major breaking changes that you need to be aware of before upgrading:
```

> The Okteto chart 1.0 comes with two breaking changes due to the following changes:

Reviewer 2: Let's be consistent in how we call it. We use ' Okteto chart 1.0', 'chart', and 'helm chart' across the doc.

> - The chart has been renamed from `okteto-enterprise` to `okteto`. This means the Builkit Kubernetes service is recreated on upgrade and Buildkit won't be available until you update your DNS entries.

Reviewer: ```suggestion
- The chart has been renamed from `okteto-enterprise` to `okteto`. This means the [BuildKit](https://github.com/moby/buildkit) Kubernetes service is recreated on upgrade and Buildkit won't be available until you update your DNS entries.
```

> - The chart has been renamed from `okteto-enterprise` to `okteto`. This means the Builkit Kubernetes service is recreated on upgrade and Buildkit won't be available until you update your DNS entries.

Reviewer 2: ```suggestion
- The helm chart has been renamed from `okteto-enterprise` to `okteto`. All the resources created by the chart will have `okteto` as their prefix instead of `okteto-enterprise`. This change will cause all resources to be recreated upon upgrade.  Depending on your cloud provider and your networking configuration, you might need to update the DNS entry of the [build service](/docs/enterprise/administration/configuration/#buildkit) after upgrading your Okteto instance.
```

> - The Okteto chart 1.0 doesn't install [cert-manager](https://cert-manager.io/) by default any more, as suggested in the [cert-manager best practices](https://cert-manager.io/docs/installation/helm/#installing-with-helm).

Reviewer 2: ```suggestion
- The Okteto chart 1.0 doesn't install [cert-manager](https://cert-manager.io/) by default any more, as suggested in the [cert-manager best practices](https://cert-manager.io/docs/installation/helm/#installing-with-helm).
```
```suggestion
- The helm chart doesn't install [cert-manager](https://cert-manager.io/) by default.
```

> To reduce the disruption of the service to a minimum, follow the following steps:

Reviewer: ```suggestion
To reduce disruption of the service to a minimum, follow the steps below:
```

> To reduce the disruption of the service to a minimum, follow the following steps:

Reviewer 2: ```suggestion
We recommend that you follow the steps below to reduce the disruption of the service to a minimum:
```

#### Before upgrade

> If you are [bringing your own certificates](/docs/enterprise/administration/certificates/), you can skip this step.

Reviewer 2: ```suggestion
You can skip this step if you are [bringing your own certificates](/docs/enterprise/administration/certificates/), 
```

> You can check if this is the case if the wildcard creation is disable in the Okteto helm values configuration:

Reviewer: ```suggestion
You can check if this is the case by seeing that the wildcard creation is disable in the Okteto helm values configuration:
```

```
wildcardCertificate:
  create: false
```

> Run the following command to make sure `helm upgrade` doesn't remove the existing cert-manager issuer and certificates:

Reviewer 2: ```suggestion
If the helm chart is generating the certificates for you,  run the following command from your command line to make sure `helm upgrade` doesn't remove the existing cert-manager issuer and certificates:
```

```
kubectl -n okteto annotate issuer okteto-okteto-enterprise helm.sh/resource-policy=keep
kubectl -n okteto annotate certificate default-ssl-certificate helm.sh/resource-policy=keep
```

> Also, add the following lines to your Okteto helm values configuration:

Reviewer 2: ```suggestion
Then, add the following configuration to your Okteto helm values file to use the existing certificate:
```

```
wildcardCertificate:
  create: false
  name: default-ssl-certificate

ingress-nginx:
  controller:
    extraArgs:
      default-ssl-certificate: $(POD_NAMESPACE)/default-ssl-certificate
```

> > Keeping the same `issuer` and `certificate` reduces moving parts during the upgrade, but we recommend to [bring your own certificate](/docs/enterprise/administration/certificates/) once the Okteto upgrade has been complete succesfully.

Reviewer: ```suggestion
> Keeping the same `issuer` and `certificate` reduces moving parts during the upgrade, but we recommend you to [bring your own certificate](/docs/enterprise/administration/certificates/) once the Okteto upgrade has been complete succesfully.
```

#### Upgrade Okteto

Run the following commands to upgrade your Okteto instance:

```
helm repo update
```

```
helm upgrade --install okteto okteto/okteto -f config.yaml --namespace=okteto
```

#### After upgrade

> You can use `kubectl` to fetch the new address that has been dynamically allocated to the Buildkit instance:

Reviewer: ```suggestion
You can use `kubectl` to fetch the new address that has been dynamically allocated to the BuildKit instance:
```

> You can use `kubectl` to fetch the new address that has been dynamically allocated to the Buildkit instance:

Reviewer 2: ```suggestion
Use `kubectl` to get the new address of the build service:
```

```console
kubectl get service -l=app.kubernetes.io/instance=okteto,app.kubernetes.io/component=buildkit --namespace=okteto
```

The output will look something like this:

```
NAME                TYPE           CLUSTER-IP      EXTERNAL-IP                           PORT(S)          AGE
okteto-buildkit     LoadBalancer   10.245.142.73   a519c8b3b27f95...elb.amazonaws.com    1234:32449/TCP   5m
```

> You'll need to take the `EXTERNAL-IP` address and add it to your DNS for the domain you have chosen to use. This is done by updating the `A` record with the name `buildkit`.

Reviewer 2: ```suggestion
Update your DNS configuration with the `EXTERNAL-IP` address.
```

> If you are [bringing your own certificates](/docs/enterprise/administration/certificates/), you should be done with the upgrade.

Reviewer 2: ```suggestion
If you are [bringing your own certificates](/docs/enterprise/administration/certificates/), your Okteto instance is now fully upgraded. The rest of the guide doesn't apply to your configuration.
```

> The Okteto upgrade might have removed cert-manager from your cluster, in case it was automatically installed by Okteto.

Reviewer 2: ```suggestion
## Install cert-manager

Okteto will remove the instance of `cert-manager` that was installed by older versions of the helm chart. If you plan on using `cert-manager` as your certificate provider, you'll need to install it manually in your cluster using the instructions below: 
```

If this was the case, you need to install cert-manager to keep renewing your certificates. To do that, execute the following commands:

```
helm repo add jetstack https://charts.jetstack.io
```

```
helm repo update
```

```

> helm install cert-manager jetstack/cert-manager --namespace=okteto --version=v1.5.2 --set global.podSecurityPolicy.enabled=true

Reviewer 2: I think we should add something like this at the end:

```
Once installed, cert-manager will continue to renew your existing certificates. Please refer to [their docs](https://cert-manager.io/docs/) if you run into any installation or certificate renewal issues.
```

Reviewer 2: do we need to give them instructions on what services to restart when a cert renews?

Author: they are restarted by `stakater`, we didn't remove that component yet
